packages/react-juce/src/lib/Stylesheet.ts

```diff
@@ -0,0 +1,177 @@
+    position?: "absolute" | "relative",
+    right?: number | string,
+    top?: number | string,
+    width?: number | string,
+}
+
+/**
+ *
+ */
+interface ImageStyles {
+    placement?: number //TODO: How to actually type this? Use enum?
+}
+
+// TODO: How to add LayoutAnimated types. Do they belong in StyleSheet
+//       or should they be an additional prop on `View` ?
+interface LayoutAnimatedProps {
+    duration?: number,
+    frameRate?: number,
+    easing?: number, //TODO: How to actually type this? Use enum?
+}
+
+interface LayoutAnimatedStyles {
+    layoutAnimated?: LayoutAnimatedProps;
+}
+
+
+/**
+ *
+ */
+interface ScrollBarStyles {
+    overflowX?: string, // TODO: Add possible strings to type
+    overflowY?: string, // TODO: Add possible strings to type
+    scrollBarColor?: string,
+    scrollBarWidth?: string | number, // TODO: Add possible strings to type
+    scrollBarTrackColor?: string,
+}
+
+interface StyleSheetObj extends BackgroundStyles, BorderStyles, FlexboxStyles, FontStyles, ImageStyles { }
+
+/**
+ *
+ */
+export interface StyleSheet {
+    [styleObj: string]: StyleSheetObj
+}
+
+/**
+ *
+ */
+export interface StyleSheetIdentifier {
+
+}
+
+export type StyleSheetIdentifierSwitch = StyleSheetIdentifier | boolean;
+
+
+// Get any css properties not beginning with a "-",
+// and build a map from any camelCase versions to
+// the hyphenated version.
+const cssPropsMap = allCssProps
+  .filter((s) => !s.startsWith("-") && s.includes("-"))
+  .reduce((acc, v) => Object.assign(acc, { [camelCase(v)]: v }), {});
+
+export default {
+    create(stylesObj: StyleSheet): StyleSheetIdentifier {
+        let nativeStyles: StyleSheet = {};
+
+        for (const subObj in stylesObj) {
+            let nativeSubObj: StyleSheetObj = {};
+
+            for (let propKey in stylesObj[subObj]) {
+                let value = stylesObj[subObj][propKey];
+
+                // If the supplied propkey is a camelCase equivalent
+                // of a css prop, first convert it to kebab-case.
+                propKey = cssPropsMap[propKey] || propKey;
+
+                // Convert provided color string to alpha-hex code for JUCE.
+                if (Colors.isColorProperty(propKey)) {
+                  value = Colors.colorStringToAlphaHex(value);
+                  if (value.startsWith("linear-gradient")) {
+                    value = Colors.convertLinearGradientStringToNativeObject(value);
+                  }
+                }
+                else if (macroPropertyGetters.hasOwnProperty(propKey)) {
+                    for (const [k, v] of macroPropertyGetters[propKey](value)) {
+                        nativeSubObj[k] = v;
+                    }
+                }
+
+                nativeSubObj[propKey] = value;
+            }
+
+            nativeStyles[subObj] = nativeSubObj;
+        }
+
+        //@ts-ignore
+        return NativeMethods.createStyleSheet(nativeStyles);
+    },
+    cx(...args: StyleSheetIdentifierSwitch[]) {
```

Comment:
Excellent thanks. I'll give this a go.

Reply:
Lol apologies as picking this up after leaving it stale for a long time. 

This is cool the only debatable issue with this approach for the `cx` function is you have to write something like this:

```
  mute_button: {
    justifyContent: "center",
    alignItems: "center",
    borderRadius: 5.0,
    borderWidth: 2.0,
    borderColor: "rgba(102, 253, 207, 1)",
    marginTop: 10,
    minWidth: 30.0,
    minHeight: 30.0,
    width: "20%",
    height: "17.5%",
  },
  mute_button_active: {
    backgroundColor: "hsla(162, 97%, 70%, 0)"
  },
  mute_button_inactive: {
    backgroundColor: "#66FDCF"
  },
  mute_button_text: {
    fontSize: 20.0,
    lineSpacing: 1.6,
    fontStyle: Text.FontStyleFlags.bold,
  },
  mute_button_text_inactive: {
      color: "#17191f"
  },
  mute_button_text_active: {
      color: "hsla(162, 97%, 70%, 1)"
  },
  
      const [muted, setMuted] = useState(false);

    const muteButtonStyles = StyleSheet.cx({[styles.mute_button]: true,
                                            [styles.mute_button_active]: muted,
                                            [styles.mute_button_inactive]: !muted});

    const muteTextStyles = StyleSheet.cx({[styles.mute_button_text]: true,
                                          [styles.mute_button_text_active]: muted,
                                          [styles.mute_button_text_inactive]: !muted});

```

In practice I'm not sure if it's a major issue but you can't rely on doing say:

```
    // Here we want to rely on positional override so that mute_button_text_active
    // overrides the default/base color in mute_button_text
    const muteTextStyles = StyleSheet.cx({[styles.mute_button_text]: true,
                                          [styles.mute_button_text_active]: muted});
```

Because you can't guarantee the ordering of the keys in the JS object/dictionary and thus you cant gurantee the order that the stylesheet objects/identifiers get passed to native in.

Reply:
(Just coming back to this PR finally, well ... again) 

I actually hit this exact case hence the comment :-) Heh. I will checkout "classnames" though. That looks great.